### PR TITLE
Rename GPUAccelerator to CUDAAccelerator

### PR DESCRIPTION
**Pull request by justusschock**
## What does this PR do?

As discussed in #13102 this renames the current GPUAccelerator to CUDAAccelerator.
Introduces `strategy='cuda'` but retains `strategy='gpu'` for this PR temporarily

In a follow-up, I will re-route `accelerator='gpu'` to support cuda and mps devices (as also discussed in #13102 )

### Does your PR introduce any breaking changes? If yes, please list them.
Yes, deprecates the current GPU Accelerator

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @justusschock @awaelchli @rohitgr7 @tchaton @kaushikb11 @akihironitta

### CODE REVIEW

File: tests/tests_pytorch/accelerators/test_accelerator_registry.py [hunk 1]
```diff
@@ -63,4 +63,4 @@ def is_available():
 
 
 def test_available_accelerators_in_registry():
-    assert AcceleratorRegistry.available_accelerators() == ["cpu", "gpu", "hpu", "ipu", "mps", "tpu"]
+    assert AcceleratorRegistry.available_accelerators() == ["cpu", "cuda", "tpu", "hpu", "ipu", "mps", "tpu"]
```
Comment (justusschock) [inline tests/tests_pytorch/accelerators/test_accelerator_registry.py:66]:
```suggestion
    assert AcceleratorRegistry.available_accelerators() == ["cpu", "cuda", "gpu", "hpu", "ipu", "mps", "tpu"]
```

File: src/pytorch_lightning/accelerators/gpu.py [hunk 2]
```diff
@@ -11,151 +11,21 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import logging
-import os
-import shutil
-import subprocess
-from typing import Any, Dict, List, Optional, Union
-
 import torch
```
Comment (justusschock) [inline src/pytorch_lightning/accelerators/gpu.py:14]:
```suggestion
```

File: docs/source-pytorch/api_references.rst [hunk 3]
```diff
@@ -12,7 +12,7 @@ accelerators
 
     Accelerator
     CPUAccelerator
-    GPUAccelerator
+    CUDAAccelerator
     HPUAccelerator
     IPUAccelerator
     TPUAccelerator
```
Comment (Borda) [inline docs/source-pytorch/api_references.rst:18]: shall be then all other updated as TPU is using XLA?

Comment (justusschock) [reply docs/source-pytorch/api_references.rst:18]: No, we just did this, because we will have multiple GPUs (currently nvidia and mps and in the future likely rocm as well). So we will have a separate accelerator for all of those and `strategy='gpu'` will still work and select among them.

Renaming TPU to XLA would only be required if there was another TPU backend

Comment (Borda) [reply docs/source-pytorch/api_references.rst:18]: I see just from esthetical point of view it is strange to have AA in sequence lol

Comment (justusschock) [reply docs/source-pytorch/api_references.rst:18]: I agree, but usually people are never expected to use these :) they go with `accelerator='cuda'` or `accelerator='gpu'` which should be fine for your aesthetics :P

File: src/pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 4]
```diff
@@ -475,7 +475,7 @@ def _map_deprecated_devices_specific_info_to_accelerator_and_device_flag(
                 if tpu_cores:
                     self._accelerator_flag = "tpu"
                 if gpus:
-                    self._accelerator_flag = "gpu"
+                    self._accelerator_flag = "cuda"
```
Comment (awaelchli) [inline src/pytorch_lightning/trainer/connectors/accelerator_connector.py:478]: accidental rename?

Comment (justusschock) [reply src/pytorch_lightning/trainer/connectors/accelerator_connector.py:478]: no that was on purpose :) Soon there will be no direct mapping of an Accelerator and the "gpu" flag anymore :)

File: src/pytorch_lightning/accelerators/cuda.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        return torch.cuda.device_count() > 0
+
+    @classmethod
+    def register_accelerators(cls, accelerator_registry: Dict) -> None:
+        accelerator_registry.register(
+            "cuda",
+            cls,
+            description=f"{cls.__class__.__name__}",
+        )
+        # temporarily enable "gpu" to point to the CUDA Accelerator
```
Comment (carmocca) [inline src/pytorch_lightning/accelerators/cuda.py:100]: Great!

**Review by carmocca — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]
LGTM! 🚀 Should we preserve the git history with "rebase&merge"? (If we do, will need to clean up the commits)

**Comment by justusschock** [comment]
@akihironitta wouldnt we just squash merge it? That should be fine, no?

File: src/pytorch_lightning/accelerators/gpu.py [hunk 6]
```diff
@@ -11,151 +11,21 @@
...
-def _get_gpu_id(device_id: int) -> str:
-    """Get the unmasked real GPU IDs."""
-    # All devices if `CUDA_VISIBLE_DEVICES` unset
-    default = ",".join(str(i) for i in range(torch.cuda.device_count()))
-    cuda_visible_devices = os.getenv("CUDA_VISIBLE_DEVICES", default=default).split(",")
-    return cuda_visible_devices[device_id].strip()
+    def __init__(self) -> None:
+        rank_zero_deprecation(
+            "The GPUAccelerator has been renamed to CUDAAccelerator and will be removed in v1.9."
+            " Please use the CUDAAccelerator instead!"
```
Comment (rohitgr7) [inline src/pytorch_lightning/accelerators/gpu.py:29]:
```suggestion
            "The `GPUAccelerator` has been renamed to `CUDAAccelerator` and will be removed in v1.9."
            " Please use the `CUDAAccelerator` instead!"
```

File: tests/tests_pytorch/deprecated_api/test_remove_1-9.py [hunk 7]
```diff
@@ -195,3 +196,13 @@ def test_pytorch_profiler_schedule_wrapper_deprecation_warning():
 def test_pytorch_profiler_register_record_function_deprecation_warning():
     with pytest.deprecated_call(match="RegisterRecordFunction` is deprecated in v1.7 and will be removed in in v1.9."):
         _ = RegisterRecordFunction(None)
+
+
+def test_gpu_accelerator_deprecation_warning():
+    with pytest.deprecated_call(
+        match=(
+            "The GPUAccelerator has been renamed to CUDAAccelerator and will be removed in v1.9."
+            + " Please use the CUDAAccelerator instead!"
```
Comment (rohitgr7) [inline tests/tests_pytorch/deprecated_api/test_remove_1-9.py:205]:
```suggestion
            "The `GPUAccelerator` has been renamed to `CUDAAccelerator` and will be removed in v1.9."
            + " Please use the `CUDAAccelerator` instead!"
```

File: src/pytorch_lightning/accelerators/__init__.py [hunk 8]
```diff
@@ -12,7 +12,7 @@
 # limitations under the License.
 from pytorch_lightning.accelerators.accelerator import Accelerator  # noqa: F401
 from pytorch_lightning.accelerators.cpu import CPUAccelerator  # noqa: F401
-from pytorch_lightning.accelerators.gpu import GPUAccelerator  # noqa: F401
+from pytorch_lightning.accelerators.cuda import CUDAAccelerator  # noqa: F401
```
Comment (rohitgr7) [inline src/pytorch_lightning/accelerators/__init__.py:15]: this breaks
```py
from pytorch_lightning.accelerators import GPUAccelerator
```
we should keep the import

**Review by rohitgr7 — COMMENTED** [review]
Agreed with @akihironitta !

we should do `rebase and merge` with 2 commits to preserve the git history. `squash and merge` will not keep it.

File: src/pytorch_lightning/accelerators/__init__.py (hunk 8 above)
Comment (justusschock) [reply src/pytorch_lightning/accelerators/__init__.py:15]: we shouldn't because then it will automatically be registered for the gpu keyword but we want it to map to the CUDAAccelerator

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/accelerators/gpu.py [hunk 9]
```diff
@@ -11,151 +11,21 @@
...
+from pytorch_lightning.utilities.rank_zero import rank_zero_deprecation
 
-import torch
 
-import pytorch_lightning as pl
-from pytorch_lightning.accelerators.accelerator import Accelerator
-from pytorch_lightning.utilities import device_parser
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from pytorch_lightning.utilities.types import _DEVICE
+class GPUAccelerator(CUDAAccelerator):
```
Comment (kaushikb11) [inline src/pytorch_lightning/accelerators/gpu.py:18]: No deprecation message for the users?

**Review by rohitgr7 — APPROVED** [review]
